### DISCUSSION TITLE
alternative way to draw lines like symbol

### DISCUSSION THREAD

**Discussion by qn222ad** [highfestiva/finplot#217] [Q&A]
I am plotting 10000 vertical lines in 100 years data and waited for 2 hours still not finish. I used adding style with this:

fplt.plot(data[f'{k}'], ax=ax, color='#1ad845', style='>', legend=f'{namer}', width=3)

which is faster but I love to see the lines, is there any alternative way of plotting lines with fplt.plot so that it can be faster?
Thanks

**Comment by highfestiva** [comment]
You're trying to scatter plot arrows (`'>'`), but you're doing it one after another instead of all at once. To see how lines are drawn, look at the btc-long-term example.

**Reply by qn222ad** [reply]
I used the add_lines in btc_longterm example, but it takes so long, wait more than 2 H, not finished, while the adding symbol, take about 10 min :(

**Reply by highfestiva** [reply]
Haha, suppose you'll have to add some rendering primitives to get that many primitives rendered fast. Add fewer lines, make them dynamic or look in to how for instance candles are rendered. Candlesticks are rendered in batch, but only a few thousand at a time, if there are too many on-screen at once, a surplus is dropped and only rendered when zooming in.

**Reply by qn222ad** [reply]
some rendering primitives to get that many primitives rendered fast... this one I will study later, they are still latin to me :).
This idea "_rendered in batch, but only a few thousand at a time, if there are too many on-screen at once, a surplus is dropped and only rendered when zooming in_." where can I find study material--- what is the main keyword?
Thanks :) In the mean time I will use plot symbol to avoid the pain of waiting:D
